### PR TITLE
os version in upgrade check

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
@rleigh-dundee asked to provide more information about OS in UpgradeCheck. This is because original implementation uses only ```platform.version()```. This change should give more details to registry.

Mac OS X:
```
>>> "%s-%s" % (platform.platform(),platform.mac_ver()[0])
'Darwin-12.5.0-x86_64-i386-64bit-10.8.5'
```

Linux:
```
>>> platform.platform()
'Linux-2.6.32-431.29.2.el6.x86_64-x86_64-with-centos-6.5-Final'
```

Windows:
```
>>> platform.platform()
'Windows-2008Server-6.0.6002-SP2'
```

Test library locally logging to web calling python directly:

```
from omero.util.upgrade_check import UpgradeCheck
check = UpgradeCheck("OMERO.test")
check.run()
```

```
2015-01-14 16:26:10,994 DEBUG [                 omero.util.UpgradeCheck] (proc.31949) run:136 Attempting to connect to http://upgrade.openmicroscopy.org.uk/?os.version=Darwin-12.5.0-x86_64-i386-64bit-10.8.5&os.arch=x86_64&version=5.1.0-m4-ice34-SNAPSHOT&python.compiler=GCC+4.2.1+Compatible+Apple+Clang+4.0+%28tags%2FApple%2Fclang-418.0.60%29&python.build=%28%27default%27%2C+%27Oct+11+2012+20%3A14%3A37%27%29&os.name=Darwin&python.version=2.7.2
```

### CODE REVIEW

File: components/tools/OmeroPy/src/omero/util/upgrade_check.py [hunk 1]
```diff
@@ -91,6 +91,17 @@ def _set(self, results, e):
         self.upgradeUrl = results
         self.exc = e
 
+    def getOSVersion(self):
+        try:
+            if len(platform.mac_ver()[0]) > 0:
+                version = "%s-%s" % (platform.platform(),
```
Comment (rleigh-dundee) [inline components/tools/OmeroPy/src/omero/util/upgrade_check.py]: I thought we were going to use `;` as the separator here so we could use a regex to separate the information?

**Comment by aleksandra-tarkowska** [comment]
@rleigh-dundee done

**Comment by rleigh-dundee** [comment]
I'm having trouble testing this; I don't get any debugging output.  How are you running the test?  i.e. working directory/environment etc.

**Comment by rleigh-dundee** [comment]
This looks fine, thanks.

For future work, unifying the reported data for both python and java would be useful; it's currently quite different.  And splitting up this information into separate columns would also make it much easier to query.

**Comment by aleksandra-tarkowska** [comment]
@rleigh-dundee as discussed these changes would require model changes in registry. I will take it to the account during upgrade.

**Comment by rleigh-dundee** [comment]
Good to merge.

**Comment by joshmoore** [comment]
Thanks, guys. Merging.

@aleksandra-tarkowska : based on what you've been looking at recently, would it make sense to backport this, or will complicate things?

**Comment by aleksandra-tarkowska** [comment]
--rebased-to #3367

**Comment by aleksandra-tarkowska** [comment]
@rleigh-dundee I forgot about that before, should Java library alter that value as well?

**Comment by rleigh-dundee** [comment]
@aleksandra-tarkowska It would certainly be nice if the Python and Java code submitted the same information, so it's meaningful to query it.

**Comment by aleksandra-tarkowska** [comment]
@joshmoore @jburel @dominikl could you also make sure java UpgradeCheck is setting right value?

**Comment by joshmoore** [comment]
The extent available in Java is:
```
>>> p = java.lang.System.getProperties()
>>> for x in p.keySet():
...     if x.startswith("os"): print x, p.get(x)
...
os.arch x86_64
os.name Mac OS X
os.version 10.10.2
```

**Comment by mtbc** [comment]
In case it's of any use, the corresponding values from Java for my laptop are, respectively,
- amd64
- Linux
- 3.2.0-4-amd64.

**Comment by rleigh-dundee** [comment]
For MacOS X that's fine.  For Linux, it's not very useful.  We're missing distribution+version; the kernel version isn't very useful in comparison.  Adding that would be very helpful; there are a few places to check to determine this e.g. `lsb-release -a`, /etc/redhat-release /etc/debian_version.  Not perfect or comprehensive, but good enough for our needs.

**Comment by mtbc** [comment]
Could `/etc/issue` be useful across distributions?

**Comment by rleigh-dundee** [comment]
Not generally, since this is intended for customisation by the admin.  The formal interface for this is `lsb-release`, but it's not necessarily available by default on all systems without LSB support being installed.  It's installed on Ubuntu by default though.  However, may be useful as a fallback in the absence of lsb-release or /etc/redhat|debian*

**Comment by mtbc** [comment]
True. I have it locally as `lsb_release`, not `lsb-release`.

**Comment by joshmoore** [comment]
cF. https://github.com/aurbroszniowski/os-platform-finder

**Pull request merged**